### DISCUSSION TITLE
What happens to a lock when the process is killed?

### DISCUSSION THREAD

**Discussion by l-salewski** [tox-dev/filelock#116] [Q&A] [answered]
First, thanks for this great library! I have a really short question.

Unfortunately I wasn't able to find anything about this in the documentation, is a lock released if the process holding it is killed by an external event? Because otherwise this might cause a deadlock situation, where the process originally holding the lock was killed and the other processes wait forever.

**Comment by gaborbernat** [comment, marked as answer]
https://py-filelock.readthedocs.io/en/latest/index.html#filelock-vs-softfilelock covers this; if the app is killed there's little we can do

**Reply by l-salewski** [reply]
Thank you for this quick answer and sorry for opening an issue about what should have been a discussion.

Thanks for pointing me to the FileLock vs SoftFileLock documentation. I am using FileLock (as all processes are running on Linux) and thus did read the [python docs](https://docs.python.org/3/library/fcntl.html#fcntl.flock) and the [unix manual](https://manpages.debian.org/bullseye/manpages-dev/flock.2.en.html).
From my small experiments with two python REPLs it seems that raised errors and keyboard interupts release the lock, such that it is accessible to other processes.

**Reply by eladn** [reply]
@l-salewski keyboard interrupts terminates the process gracefully - the builtin python's signal handler catches this and is responsible of releasing the allocated resources. I'm still not sure about what happens if it's forcefully killed with SIGKILL.
